source/extensions/common/wasm/wasm.h

```diff
@@ -427,21 +453,19 @@ class Wasm : public Envoy::Server::Wasm,
              public Logger::Loggable<Logger::Id::wasm>,
              public std::enable_shared_from_this<Wasm> {
 public:
-  Wasm(absl::string_view vm, absl::string_view id, absl::string_view vm_configuration,
-       Upstream::ClusterManager& cluster_manager, Event::Dispatcher& dispatcher,
-       Stats::Scope& scope, envoy::api::v2::core::TrafficDirection direction,
-       const LocalInfo::LocalInfo& local_info,
-       const envoy::api::v2::core::Metadata* listener_metadata,
-       Stats::ScopeSharedPtr owned_scope = nullptr);
+  Wasm(absl::string_view vm, absl::string_view vm_id, absl::string_view vm_configuration,
+       PluginSharedPtr plugin, Upstream::ClusterManager& cluster_manager,
+       Event::Dispatcher& dispatcher);
   Wasm(const Wasm& other, Event::Dispatcher& dispatcher);
   ~Wasm() {}
 
-  bool initialize(const std::string& code, absl::string_view name, bool allow_precompiled);
+  bool initialize(const std::string& code, bool allow_precompiled = false);
   bool configure(Context* root_context, absl::string_view configuration);
-  Context* start(absl::string_view root_id,
-                 absl::string_view vm_configuration); // returns the root Context.
+  Context* start(); // returns the root Context.
 
-  absl::string_view id() const { return id_; }
+  absl::string_view vm_id() const { return vm_id_; }
+  absl::string_view configured_vm_id() const { return configured_vm_id_; }
```

Comment:
What's the point of `configured_vm_id`? I don't see it being used anywhere, and it's always the same value as `vm_id`.

Reply:
Done